data/systemd/Makefile

```diff
@@ -19,6 +19,9 @@ BINDIR := /usr/bin
 LIBEXECDIR := /usr/lib
 SYSTEMDSYSTEMUNITDIR := /lib/systemd/system
 
+# NOTE: the code in wrappers/core18.go assumes that all our service units's
+# name start with the "snapd." prefix; units whose filename does not have this
+# prefix will not be installed on the host!
 SYSTEMD_UNITS_GENERATED := $(wildcard *.in)
 # NOTE: sort removes duplicates so this gives us all the units, generated or otherwise
 SYSTEMD_UNITS = $(sort $(SYSTEMD_UNITS_GENERATED:.in=) $(wildcard *.service) $(wildcard *.timer) $(wildcard *.socket))
```

Comment:
If it only makes sense to install systemd units that begin with `snapd.`, would it be worth adjusting the globs here to match `snapd.*.in`, etc? That might help make the problem apparent even earlier.

Reply:
Yes, it does, but then I cannot merge this with the skip-spread label :-D